backstory/src/test/java/com/google/sps/images/VisionImagesManagerTest.java

```diff
@@ -69,6 +111,58 @@ public void testCorrectRequestInput() throws IOException {
     ArgumentCaptor<List<AnnotateImageRequest>> argument = ArgumentCaptor.forClass(List.class);
     verify(mockImageAnnotatorClient).batchAnnotateImages(argument.capture());
     List<AnnotateImageRequest> requests = argument.getValue();
-    Assert.assertEquals(requests.size(), 1);
+    Assert.assertEquals(rawImageDataList.size(), requests.size());
+
+    // Check that features we want are requested
+    List<Feature> expectedFeatures = new ArrayList<Feature>();
+
+    Feature labelFeature = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
+    Feature landmarkFeature = Feature.newBuilder().setType(Feature.Type.LANDMARK_DETECTION).build();
+
+    expectedFeatures.add(labelFeature);
+    expectedFeatures.add(landmarkFeature);
+    
+    for (AnnotateImageRequest request: requests) {
+      Assert.assertEquals(expectedFeatures, request.getFeaturesList());
+    }
```

Comment:
This would be clearer (less logic):
```
Assert.assertEquals(1, requests.size());
AnnotateImageRequest request = requests.get(0);
Assert.assertEquals(expectedFeatures, request.getFeaturesList());
```

Reply:
Thanks! I'll replace the existing code with this.

Reply:
Done